File: go/lasagna-master/lasagna-master.go
package lasagna

func PreparationTime(layers []string, avgPerLayer int) int {
    if avgPerLayer == 0 {
        avgPerLayer = 2
    }
    return len(layers) * avgPerLayer
}

func Quantities(layers []string) (int, float64) {
    noodles, sauce := 0, float64(0)
    for _, layer := range layers {
        switch layer {
            case "noodles":
        		noodles += 50
            case "sauce":
        		sauce += 0.2
        }
    }
	return noodles, sauce
}

func AddSecretIngredient(friendsList []string, myList []string) []string {
    return append(myList, friendsList[len(friendsList)-1])
}

func ScaleRecipe(list []float64, portions int) []float64 {
	output := make([]float64, len(list))
	for i := 0; i < len(list); i++ {
		output[i] = list[i] * float64(portions) / 2
	}
	return output
}

File: javascript/lasagna-master/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

export function cookingStatus(remainingTime) {
  if (remainingTime === 0) {
    return "Lasagna is done.";
  } else if (remainingTime > 0) {
    return "Not done, please wait.";
  }
  return "You forgot to set the timer.";
}

export function preparationTime(layers, timePerLayer=2) {
    return layers.length * timePerLayer;
}

export function quantities(ingredients) {
    let  [sauce, noodles] = [0,0]
    for (let ingredient of ingredients) {
        switch(ingredient) {
            case "sauce":
                sauce += 0.2;
                break;
            case "noodles":
                noodles += 50;
                break;
        }
    }
    return {sauce, noodles};
}

export function addSecretIngredient(friendsList, myList) {
    myList.push(friendsList[friendsList.length - 1])
}

export function scaleRecipe(recipe, factor) {
    let scaledRecipe = {};
    for (let ingredient in recipe) {
        scaledRecipe[ingredient] = (recipe[ingredient] * factor)/2;
    }
    return scaledRecipe;
}
